rust/src/ssh/ssh.rs

```diff
@@ -179,8 +212,24 @@ impl SSHState {
                             let remlen = rem.len() as u32;
                             hdr.record_left = head.pkt_len - 2 - remlen;
                             //header with rem as incomplete data
-                            if head.msg_code == SSH_MSG_NEWKEYS {
-                                hdr.flags = SSHConnectionState::SshStateFinished;
+                            match head.msg_code { 
+                                parser::MessageCode::SshMsgNewKeys => {
+                                    hdr.flags = SSHConnectionState::SshStateFinished;
+                                }
+                                parser::MessageCode::SshMsgKexinit if HASSH_ENABLED.load(Ordering::Relaxed) => {
+                                    // check if buffer is bigger than maximum reassembled packet size
+                                    if hdr.record_left < SSH_MAX_REASSEMBLED_RECORD_LEN as u32 {
+                                        // saving type of incomplete kex message
+                                        hdr.record_left_msg = parser::MessageCode::SshMsgKexinit;
+                                        return AppLayerResult::incomplete(SSH_RECORD_HEADER_LEN as u32, hdr.record_left as u32);
+                                    }
+                                    else {
+                                        SCLogDebug!("SSH buffer is bigger than maximum reassembled packet size");
+                                        self.set_event(SSHEvent::InvalidRecord);
+                                        return AppLayerResult::err();
```

Comment:
Why return error here ?
We can continue parsing by skipping this big record

Reply:
Should it return just `return AppLayerResult::ok()` ?

Reply:
There should not be any `return` at all here, and let the function continue to the `return AppLayerResult::ok();` present a few lines below